### PR TITLE
Updated to Swift 4.1

### PR DESCRIPTION
**Pull request by Davidde94**
All deprecation warnings have been silenced by updating to the latest Swift 4.1 syntax.

### CODE REVIEW

**Comment by ianpartridge** [comment]
Thanks for the PR 👍 

If possible we would like to keep support for Swift 4.0 as well - is that possible?

**Comment by Davidde94** [comment]
@ianpartridge should be - let me have a play and send a few more commits

**Comment by Davidde94** [comment]
@ianpartridge I've added support for 4.0, but I'm not 100% sure how to specific multiple support swift versions using the .swift_version file. Any ideas?

File: .travis.yml [hunk 1]
```diff
@@ -17,8 +17,11 @@ matrix:
       dist: trusty
       sudo: required
     - os: osx
-      osx_image: xcode9.2
-      sudo: required
+        osx_image: xcode9.2
+        sudo: required
```
Comment (ianpartridge) [inline .travis.yml]: Add a line below this:  `env: SWIFT_SNAPSHOT=4.0.3`

File: README.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ MySQL plugin for Swift-Kuery framework
 [MySQL](https://dev.mysql.com/) plugin for the [Swift-Kuery](https://github.com/IBM-Swift/Swift-Kuery) framework. It enables you to use Swift-Kuery to manipulate data in a MySQL database.
 
 ## Swift version
-The latest version of SwiftKueryMySQL requires **Swift 4.0**. You can download this version of the Swift binaries by following this [link](https://swift.org/download/). Compatibility with other Swift versions is not guaranteed.
+The latest version of SwiftKueryMySQL requires **Swift 4.1**. You can download this version of the Swift binaries by following this [link](https://swift.org/download/). Compatibility with other Swift versions is not guaranteed.
```
Comment (ianpartridge) [inline README.md]: Swift 4.0 or later.

**Comment by ianpartridge** [comment]
`.swift-version` should be the latest version supported so you're good.

**Comment by Davidde94** [comment]
Great, thanks 👍

**Comment by Davidde94** [comment]
Any update here?

**Comment by ianpartridge** [comment]
Sorry for the delay - could you make the 2 small updates I requested?

**Comment by EnriqueL8** [comment]
Hey @Davidde94, any news on this?

**Comment by Davidde94** [comment]
Shit sorry guys - I'll get this done tomorrow!

**Comment by EnriqueL8** [comment]
@Davidde94 We just merged an update for `4.1.2`, could you rebase this PR on top of master? We want to merge the fixes for the warnings

**Comment by Davidde94** [comment]
@EnriqueL8 SO SORRY! I will actually get this done today!

**Comment by Davidde94** [comment]
@EnriqueL8 @ianpartridge done, I think

File: docker-compose.yml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 test:
-  image: ibmcom/swift-ubuntu:4.0.3
+  image: ibmcom/swift-ubuntu:4.1.1
```
Comment (EnriqueL8) [inline docker-compose.yml]: This would need to be updated to the latest tag - `4.1.2`

**Comment by Davidde94** [comment]
@EnriqueL8 @ianpartridge I've made the requested changes :)

**Pull request merged**